### ISSUE TITLE
Suggestion about motifs.create()

### ISSUE THREAD

**Issue by shenwei356** [biopython/biopython#476]
#### Advice on method `motifs.create`

It only accept list

```
seqs = seq_iter(args.infile)    # an iter
seqs2 = [seq for seq in seqs]   # have to do this
m = motifs.create(seqs2)        # accept list as arguments
```

If I wrote:

```
seqs = seq_iter(args.infile)    # an iter
m = motifs.create(seqs) 
```

Error comes out:

```
  File "seq_pwm.py", line 62, in <module>
    print(m.pwm)
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/__init__.py", line 320, in pwm
    return self.counts.normalize(self._pseudocounts)
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/matrix.py", line 279, in normalize
    return PositionWeightMatrix(self.alphabet, counts)
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/matrix.py", line 289, in __init__
    self[letter][i] /= total
ZeroDivisionError: float division by zero
```

Here is the cause. In the `__init__` method of class Bio.Motif.Instances, `instances` is read two times with `for instance in instances`. But for iterator, It could only be read once.

We can use itertools.tee() function to create a backup generator (http://stackoverflow.com/questions/1271320/reseting-generator-object-in-python)

```
y = FunctionWithYield()
y, y_backup = tee(y)
for x in y:
    print(x)
for x in y_backup:
    print(x)
```

In the current branch of Motif code (https://github.com/biopython/biopython/blob/master/Bio/Motif/_Motif.py#L22), The code seems change a lot, I didn't test it.

Thank you all

**Comment by mdehoon** [comment]
What happens if you call motifs.create(seqs) ?

**Comment by shenwei356** [comment]
@mdehoon, here is the error when using ```motifs.create(seqs) ```



```
Traceback (most recent call last):
  File "seq_pwm.py", line 59, in <module>                                                                              
    m = motifs.create(seqs)                                                                                            
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/__init__.py", line 26, in create
    return Motif(instances=instances, alphabet=alphabet)
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/__init__.py", line 238, in __init__
    counts = self.instances.count()
  File "/usr/lib64/python3.4/site-packages/Bio/motifs/__init__.py", line 193, in count
    counts[letter] = [0] * self.length
TypeError: can't multiply sequence by non-int of type 'NoneType'
```

The seq_iter function:

```
def seq_iter(file):
    if file:
        found = re.search(r'(?i)(fasta|fa|fastq|fq)(.gz)?$', file)
        if not found:
            print("invalid file name suffix.\nfile name should like this: infile.[fasfa|fa|fastq|fq][.gz]", file=sys.stderr)
            sys.exit(1)
        seq_format, is_gz = found.groups()
        if seq_format == 'fa':
            seq_format = 'fasta'
        if seq_format == 'fq':
            seq_format = 'fastq'

        fh = gzip.open(file, 'rt') if is_gz else open(file, 'r')
        for record in SeqIO.parse(fh, seq_format):
            yield record.seq
        fh.close()
    else:
        for line in sys.stdin:
            yield Seq(line.strip())
```

**Comment by mdehoon** [comment]
Can you check line 163 of Bio/motifs/__init__.py :

self.length = len(instance)

The length should have been initialized there.

**Comment by shenwei356** [comment]
@mdehoon I've update this issue.

**Comment by peterjc** [comment]
A shorter workaround might be to call ``list(...)`` rather than doing a trivial list comprehension:

```
seqs = seq_iter(args.infile)    # an iter
m = motifs.create(list(seqs))
```

**Comment by shenwei356** [comment]
I'm thinking generator could save memory.

**Comment by mdehoon** [comment]
Fixed by #3334
